src/Umbraco.Web.BackOffice/Controllers/TrackedReferencesController.cs

```diff
@@ -0,0 +1,144 @@
+            };
+
+            var relations = _relationService.GetPagedParentEntitiesByChildId(id, pageNumber - 1, pageSize, out var totalRecords, relationTypes, objectType);
+
+            return new PagedResult<EntityBasic>(totalRecords, pageNumber, pageSize)
+            {
+                Items = relations.Cast<ContentEntitySlim>().Select(rel => new EntityBasic
+                {
+                    Id = rel.Id,
+                    Key = rel.Key,
+                    Udi = Udi.Create(udiType, rel.Key),
+                    Icon = rel.ContentTypeIcon,
+                    Name = rel.Name,
+                    Alias = rel.ContentTypeAlias
+                })
+            };
+        }
+
+        [HttpGet]
+        public bool HasReferencesInDescendants(int id, string entityType)
+        {
+            var currentEntity = _entityService.Get(id);
+
+            if (currentEntity != null)
+            {
+                var currentObjectType = ObjectTypes.GetUmbracoObjectType(currentEntity.NodeObjectType);
+
+                var objectType = ObjectTypes.GetUmbracoObjectType(entityType);
+                var relationTypes = new string[]
+                {
+                    Constants.Conventions.RelationTypes.RelatedDocumentAlias,
+                    Constants.Conventions.RelationTypes.RelatedMediaAlias
+                };
+
+                var pageSize = 1000;
+                var currentPage = 0;
+
+                var entities = new List<IEntitySlim>();
+
+                do
+                {
+                    entities = _entityService.GetPagedDescendants(id, currentObjectType, currentPage, pageSize, out _).ToList();
+
+                    var ids = entities.Select(x => x.Id).ToArray();
+
+                    var relations = _relationService.GetPagedParentEntitiesByChildIds(ids, 0, int.MaxValue, out _, relationTypes, objectType);
+
+                    if (relations.Any())
+                    {
+                        return true;
+                    }
+
+                    currentPage++;
+
+                } while (entities.Count == pageSize);
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        ///     Get entities of the items in relation from selected integer ids
+        /// </summary>
+        /// <param name="ids"></param>
+        /// <param name="entityType"></param>
+        /// <param name="pageNumber"></param>
+        /// <param name="pageSize"></param>
+        /// <returns></returns>
+        /// <remarks>
+        ///     We allow for POST because there could be quite a lot of ids
+        /// </remarks>
+        [HttpGet]
+        [HttpPost]
+        public PagedResult<EntityBasic> CheckLinkedItems([FromJsonPath] int[] ids, string entityType, int pageNumber = 1, int pageSize = 100)
+        {
+            if (pageNumber <= 0 || pageSize <= 0)
+            {
+                throw new NotSupportedException("Both pageNumber and pageSize must be greater than zero");
+            }
```

Comment:
```suggestion
        public ActionResult<PagedResult<EntityBasic>> CheckLinkedItems([FromJsonPath] int[] ids, string entityType, int pageNumber = 1, int pageSize = 100)
        {
            if (pageNumber <= 0 || pageSize <= 0)
            {
                return BadRequest("Both pageNumber and pageSize must be greater than zero");
            }
```

Should not throw from actions. And always return `ActionResult<T>`